File: EMS-backend/src/main/java/com/hogwarts/EmManagebackend/service/TodoService.java
package com.hogwarts.EmManagebackend.service;

import com.hogwarts.EmManagebackend.dto.TodoDto;

public interface TodoService {
    TodoDto findByEmployeeIdAndTodoId(Long employeeId, Long todoId);

    TodoDto createTodo(Long employeeId, TodoDto todoDto);

    TodoDto updateTodo(Long employeeId, Long todoId, TodoDto todoDto);

    void deleteById(Long employeeId, Long id);

    TodoDto completeTodo(Long id);

    TodoDto inCompleteTodo(Long id);

    TodoDto inProgressTodo(Long id);

}

File: EMS-frontend/src/services/TodoService.js
import axios from "axios";

const TODO_REST_API_BASE = 'http://localhost:8080/api/employees/todos';

export const listEmployeeTodo = (employeeId) => axios.get(TODO_REST_API_BASE + "/" + employeeId + "/lists");

export const addTodo = (employeeId, todo) => axios.post(TODO_REST_API_BASE + "/" + employeeId + "/add", todo);

export const updateTodo = (employeeId, todoId, todo) => axios.put(TODO_REST_API_BASE + "/" + employeeId + "/update/" + todoId, todo);

export const deleteTodo = (employeeId, todoId) => axios.delete(TODO_REST_API_BASE + "/" + employeeId + "/delete/" + todoId);

export const completeTodo = (todoId) => axios.patch(TODO_REST_API_BASE + "/" + todoId + '/complete');

export const inCompleteTodo = (todoId) => axios.patch(TODO_REST_API_BASE + "/" + todoId + '/incomplete');

export const inProgressTodo = (todoId) => axios.patch(TODO_REST_API_BASE + "/" + todoId + '/progress');

export const getTodo = (employeeId, todoId) => axios.get("http://localhost:8080/api/employees/todos/" + employeeId + "/todo/" + todoId);



